File: search/linear_search/java/LinearSearch.java

public class LinearSearch {
	
	// Search through array linearly, returns index if found, -1 if not found
	public static int search(int[] array, int key) {
		int index = -1;
		
		for (int i = 0; i < array.length; i++) {
			if(array[i] == key) return i;
		}
		
		return index;
	}

	public static int searchWithRecursion(int [] array, int key, int index) {
		if(index == array.length) {
			return -1;
		}

		if(array[index] != key) {
			searchWithRecursion(array, key, index+1);
		} else {
			return index;
		}
	}
	
	public static void main(String[] args) {
		int[] array = {1,2,3,4,5};
		int ans = LinearSearch.search(array, 6);
		System.out.println(ans);

		int ansWithRecursion = LinearSearch.searchWithRecursion(array, 4, 0);
		System.out.println(ansWithRecursion);
	}
}

File: search/linear_search/c/linear search.c
#include<stdio.h>
int main()
{
    int i,j,n;
     printf("size of array - ");
    scanf("%d",&n);

    int a[n];
    int c=0;

    printf("enter array elements \n");
    for(i=0; i<n; i++)
        scanf("%d",&a[i]);

    printf("element to search - ");
    scanf("%d",&j);
    for(i=0; i<n; i++)
    {
        if(a[i]==j)
        {
            c=1;
            break;
        }
    }
    if(i<n)
        printf("element found at index %d",i);
    else
        printf("element not found");
}
